### ISSUE TITLE
[Question] Predicted action probabilities

### ISSUE THREAD

**Issue by pookiee** [takuseno/d3rlpy#348]
algorithm.predict() can be used to return actions. Does algorithm.predict_value(observation, action = np.array([1]) ) return the probability of choosing action 1? If not, how can the action probabilities be extracted (e.g., rescaling the action values to sum to 1, get some other output, etc...)?

**Comment by takuseno** [comment]
@pookiee Thanks for the issue and sorry for the late reply. Action probability is a little vague word here because, assuming that you're using a discrete action-space algorithm such as DQN, the actions are selected based on the epsilon-greedy strategy. In this case, the action probability cannot be simply computed with action-values. If you can provide more contexts, I can follow up.

**Comment by pookiee** [comment]
@takuseno 
Apologies for the vagueness. Yes, for discrete-action space algorithms. This is mainly for inference purposes, so hopefully not having to deal with epsilon-greedy strategies.
I have previously used Ray RLlib which has a convenient function to extract action probabilities from the policy of the algorithm [(algorithm.compute_single_action())](https://discuss.ray.io/t/how-do-you-get-action-probabilities-from-a-policy/5712/2). From my understanding, this is done by applying softmax to the action-values (which have some extra processing since they are logits) though I am not certain. This method can be applied to their DQN algorithms (and all others). Ultimately, I could rerun your algorithm.predict_value() for the different actions and then run softmax (with temperature if the algorithm uses it) to find the probabilities, but again, I am not quite certain of this approach.

**Comment by takuseno** [comment]
Thanks for the explanation. Hmm, I see. What you could do is to compute softmax over action-values over all actions. I need to dig into how RLlib computes this though.

**Comment by takuseno** [comment]
Let me close this issue. Feel free to reopen this if there is any further discussion.
